ivy/functional/backends/torch/gradients.py

```diff
@@ -31,15 +35,25 @@ def execute_with_gradients(func, xs, retain_grads=False):
         y = func_ret
         rest = tuple()
     y = ivy.to_native(y)
-    x_grads_flat = list(
-        torch.autograd.grad(
-            [y],
-            [v for k, v in xs.to_iterator()],
+    if isinstance(xs, ivy.Container):
+        x_grads_flat = [
+            torch.autograd.grad(
+                y[key],
```

Comment:
we should assume that `y` is always single array, and never a container, you can use the original code unmodified in cases where `xs` is a container for all backends.